src/test/java/uk/gov/companieshouse/api/testdata/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,213 @@
+        boolean userExists = userServiceImpl.userExists("userId");
+
+        assertTrue(userExists, "User should exist");
+        verify(userRepository, times(1)).findById("userId");
+    }
+
+    @Test
+    void testUserDoesNotExist() {
+        when(userRepository.findById(any(String.class))).thenReturn(Optional.empty());
+
+        boolean userExists = userServiceImpl.userExists("userId");
+
+        assertFalse(userExists, "User should not exist");
+        verify(userRepository, times(1)).findById("userId");
+    }
+
+    @Test
+    void testDeleteUser() throws DataException {
+        Users mockUser = new Users();
+        mockUser.setId("userId");
+
+        when(userRepository.findById("userId")).thenReturn(Optional.of(mockUser));
+        doNothing().when(userRepository).delete(any(Users.class));
+
+        assertTrue(userServiceImpl.userExists("userId"), "User should exist before deletion");
+
+        userServiceImpl.delete("userId");
+
+        verify(userRepository, times(1)).delete(mockUser);
+    }
+
+    @Test
+    void testDeleteUserNotFound() {
+        when(userRepository.findById("userId")).thenReturn(Optional.empty());
+
+        assertFalse(userServiceImpl.userExists("userId"), "User should not exist before deletion");
+
+        assertThrows(DataException.class, () -> userServiceImpl.delete("userId"));
+    }
+
+    @Test
+    void testDeleteUserWithRoles() throws DataException {
+        String userId = "userId";
+        Users mockUser = new Users();
+        mockUser.setId(userId);
+        mockUser.setRoles(List.of("role1", "role2"));
+
+        Roles mockRole1 = new Roles();
+        mockRole1.setId("role1");
+        Roles mockRole2 = new Roles();
+        mockRole2.setId("role2");
+
+        when(userRepository.findById(userId)).thenReturn(Optional.of(mockUser));
+        when(roleRepository.findById("role1")).thenReturn(Optional.of(mockRole1));
+        when(roleRepository.findById("role2")).thenReturn(Optional.of(mockRole2));
+        doNothing().when(roleRepository).delete(any(Roles.class));
+        doNothing().when(userRepository).delete(any(Users.class));
+
+        userServiceImpl.delete(userId);
+
+        verify(roleRepository, times(1)).delete(mockRole1);
+        verify(roleRepository, times(1)).delete(mockRole2);
+        verify(userRepository, times(1)).delete(mockUser);
+    }
+
+    @Test
+    void testDeleteUserThrowsException() {
+        String userId = "userId";
+        when(userRepository.findById(userId)).thenThrow(new RuntimeException("Database error"));
+
+        DataException exception = assertThrows(DataException.class, () -> userServiceImpl.delete(userId));
+
+        assertEquals("Failed to delete user", exception.getMessage());
+    }
+
+    @Test
+    void testGenerateRandomString() throws Exception {
+        Method method = UserServiceImpl.class.getDeclaredMethod("generateRandomString", int.class);
```

Comment:
I don't think we need unit test for this private method

Reply:
removed this.